src/content_cao.cpp

```diff
@@ -716,12 +716,26 @@ void GenericCAO::updateLightNoCheck(u8 light_at_pos)
 	}
 }
 
-v3s16 GenericCAO::getLightPosition()
+m4x4f GenericCAO::getTransform()
 {
-	if (m_is_player)
-		return floatToInt(m_position + v3f(0, 0.5 * BS, 0), BS);
+	scene::ISceneNode *node = getSceneNode();
+	if (node)
+		return m4x4f(node->getAbsoluteTransformation());
+	
+	return m4x4f( m4x4f::EM4CONST_IDENTITY );
+}
 
-	return floatToInt(m_position, BS);
+v3s16 GenericCAO::getLightPosition()
+{
+	// Players and entities now both use a property to define the light sampling offset.
+	// The offset is in the entity's local space.
+	
+	v3f offset = v3f( m_prop.light_anchor );
```

Comment:
Neither place a space after `(` nor before `)`/`,`, same for the occurrences below.